src/main/kotlin/com/devh/vitstore/service/product/ProductService.kt

```diff
@@ -0,0 +1,30 @@
+package com.devh.vitstore.service.product
+
+import com.devh.vitstore.model.product.ProductDao
+import com.devh.vitstore.model.product.ProductDto
+import com.devh.vitstore.repository.ProductRepository
+import javassist.NotFoundException
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import java.util.NoSuchElementException
+import javax.transaction.Transactional
+
+@Service
+@Transactional
+class ProductService {
+    @Autowired
+    private lateinit var productRepository: ProductRepository
+
+    fun findOneById(id: Long): ProductDao? {
+        return productRepository.findById(id).orElseThrow { NoSuchElementException() }
```

Comment:
`ProductDao?` có xóa `ProductDao` vì khi ko tìm thấy đã raise exception rồi
`productRepository.findById(id) ?: throw NotFoundException("Product not found with id: $id")`

Reply:
@limkimhuor cái fun findById(id) là hàm có sẵn, và nó trả về kiểu Optional<ProductDao?> anh ạ, nên ko bỏ ? đi đc đâu